src/main/java/com/pinterest/secor/parser/PartitionFinalizer.java

```diff
@@ -85,8 +85,19 @@ public PartitionFinalizer(SecorConfig config) throws Exception {
         return mMessageParser.getFinalizedUptoPartitions(lastMessages, committedMessages);
     }
 
-    private void finalizePartitionsUpTo(String topic, String[] uptoPartitions) throws Exception {
-        final String s3Prefix = "s3n://" + mConfig.getS3Bucket() + "/" + mConfig.getS3Path();
+    private void finalizePartitionsUpTo(String topic, String[] uptoPartitions) throws Exception { 
+        String prefix = "";
+        if (mConfig.getCloudService() != null && mConfig.getCloudService().equals("Swift")) { 
```

Comment:
I thought that cloud.service must be set. In that case it's OK to fail with NullPointerException. Remove mConfig.getCloudService() != null.